sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -181,16 +181,31 @@ case class MapKeys(child: Expression)
   """,
   group = "array_funcs",
   since = "2.4.0")
-case class ArraysZip(children: Seq[Expression]) extends Expression with ExpectsInputTypes {
+case class ArraysZip(children: Seq[Expression], names: Seq[String])
+  extends Expression with ExpectsInputTypes {
+
+  def this(children: Seq[Expression]) = {
+    this(
+      children,
+      children.zipWithIndex.map {
+        case (u: UnresolvedAttribute, _) => u.nameParts.last
+        case (a: Alias, _) => a.name
+        case (a: Attribute, _) => a.name
+        case (_, idx) => idx.toString
```

Comment:
Shall we add  `_` for column name `idx` when the children are other expressions? e.g. `_1`,` _2`. As I know Spark auto-generate schema always starts with an underline, such as read a CSV file without headers.

Reply:
Thank you for your suggestion but I don't think we should do it at least in this PR because it breaks [compatibility](https://github.com/apache/spark/blob/a7369b3080ec3d76957df63cf905a68e41197ba3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L193)

Reply:
make sense